packages/debounce/src/debounce.ts

```diff
@@ -28,10 +28,9 @@ interface DebounceOptions {
   trailing?: boolean
 }
 
-export type Debounced = {
-  (...args: unknown[]): unknown
+export type Debounced<F extends (...args: any) => any> = F & {
   cancel: () => void
-  flush: () => void
```

Comment:
Also I've fixed the type here. Now when you use this function it will correctly have the same parameter types and return types as the  function debounced with the help of some TS generics trickery.